### PR TITLE
fix: explicitly delete collection of catalog source pods in grpc e2e test

### PR DESCRIPTION
**Pull request by exdx**
**Description of the change:**

**Motivation for the change:**
```
 • Failure [40.912 seconds]
Catalog
/home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/catalog_e2e_test.go:33
  delete gRPC registry pod triggers recreation [It]
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/catalog_e2e_test.go:614

  
  	Error Trace:	catalog_e2e_test.go:679
  	            				runner.go:113
  	            				runner.go:64
  	            				it_node.go:26
  	            				spec.go:215
  	            				spec.go:138
  	            				spec_runner.go:200
  	            				spec_runner.go:170
  	            				spec_runner.go:66
  	            				suite.go:62
  	            				ginkgo_dsl.go:226
  	            				ginkgo_dsl.go:214
  	            				e2e_test.go:54
  	Error:      	Received unexpected error:
  	            	pods "catalog-ktxs2-6c2hs" not found
```
**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: test/e2e/catalog_e2e_test.go [hunk 1]
```diff
@@ -676,7 +677,25 @@ var _ = Describe("Catalog", func() {
 		require.NoError(GinkgoT(), err)
 
 		// Delete the registry pod
-		require.NoError(GinkgoT(), c.KubernetesInterface().CoreV1().Pods(testNamespace).Delete(context.TODO(), name, metav1.DeleteOptions{}))
+		Eventually(func() error {
```
Comment (benluddy) [inline test/e2e/catalog_e2e_test.go:680]: While you are editing this test, can you clean up the delete behavior at L649 (to make it retry failures and accept NotFound errors)?

Comment (exdx) [reply test/e2e/catalog_e2e_test.go:680]: In reality we don't even need this anymore since after `It` spec we run the test cleaner

**Comment by exdx** [comment]
Also removed explicit cleanup from this test since this is now covered by the updated `AfterEach` cleanup that runs after every `It` spec in this e2e suite

**Pull request merged**